From 83665d8452dbc6976151f68c7ac205f5b7f9d936 Mon Sep 17 00:00:00 2001
From: Leo Hsu <leohsu@google.com>
Date: Tue, 2 Oct 2018 09:57:03 +0800
Subject: [PATCH 37/70] RESTRICT AUTOMERGE: Removes feature flag for faster
 emergency call.

Removes feature flag and enables faster emergency call features by
default.

Test: Manually

Bug: 112168722
Bug: 117128459
Change-Id: I07fa71c774a2cd60e8f046fbb9acac3012fa862e
---
 src/com/android/phone/EmergencyDialer.java | 48 ++++------------------
 1 file changed, 8 insertions(+), 40 deletions(-)

diff --git a/src/com/android/phone/EmergencyDialer.java b/src/com/android/phone/EmergencyDialer.java
index b9a64719e..ddc6f09dc 100644
--- a/src/com/android/phone/EmergencyDialer.java
+++ b/src/com/android/phone/EmergencyDialer.java
@@ -62,7 +62,6 @@ import android.text.TextUtils;
 import android.text.TextWatcher;
 import android.text.method.DialerKeyListener;
 import android.text.style.TtsSpan;
-import android.util.FeatureFlagUtils;
 import android.util.Log;
 import android.util.TypedValue;
 import android.view.HapticFeedbackConstants;
@@ -213,9 +212,6 @@ public class EmergencyDialer extends Activity implements View.OnClickListener,
 
     private static final int BAD_EMERGENCY_NUMBER_DIALOG = 0;
 
-    /** 90% opacity, different from other gradients **/
-    private static final int BACKGROUND_GRADIENT_ALPHA = 230;
-
     /** 85% opacity for black background **/
     private static final int BLACK_BACKGROUND_GRADIENT_ALPHA = 217;
 
@@ -262,8 +258,6 @@ public class EmergencyDialer extends Activity implements View.OnClickListener,
     private boolean mIsWfcEmergencyCallingWarningEnabled;
     private float mDefaultDigitsTextSize;
 
-    private boolean mAreEmergencyDialerShortcutsEnabled;
-
     private MetricsWriter mMetricsWriter;
     private SensorManager mSensorManager;
     private Sensor mProximitySensor;
@@ -330,22 +324,11 @@ public class EmergencyDialer extends Activity implements View.OnClickListener,
 
         getWindow().setAttributes(lp);
 
-        mAreEmergencyDialerShortcutsEnabled = FeatureFlagUtils
-                .isEnabled(this, FeatureFlagUtils.EMERGENCY_DIAL_SHORTCUTS);
-        Log.d(LOG_TAG, "Enable emergency dialer shortcut: "
-                + mAreEmergencyDialerShortcutsEnabled);
-
         mColorExtractor = new ColorExtractor(this);
 
         // It does not support dark text theme, when emergency dialer shortcuts are enabled.
         // And the background color is black with 85% opacity.
-        if (mAreEmergencyDialerShortcutsEnabled) {
-            updateTheme(false);
-        } else {
-            GradientColors lockScreenColors = mColorExtractor.getColors(WallpaperManager.FLAG_LOCK,
-                    ColorExtractor.TYPE_EXTRA_DARK);
-            updateTheme(lockScreenColors.supportsDarkText());
-        }
+        updateTheme(false);
 
         setContentView(R.layout.emergency_dialer);
 
@@ -363,8 +346,7 @@ public class EmergencyDialer extends Activity implements View.OnClickListener,
         ((WindowManager) getSystemService(Context.WINDOW_SERVICE))
                 .getDefaultDisplay().getSize(displaySize);
         mBackgroundGradient.setScreenSize(displaySize.x, displaySize.y);
-        mBackgroundGradient.setAlpha(mAreEmergencyDialerShortcutsEnabled
-                ? BLACK_BACKGROUND_GRADIENT_ALPHA : BACKGROUND_GRADIENT_ALPHA);
+        mBackgroundGradient.setAlpha(BLACK_BACKGROUND_GRADIENT_ALPHA);
         getWindow().setBackgroundDrawable(mBackgroundGradient);
 
         // Check for the presence of the keypad
@@ -430,10 +412,8 @@ public class EmergencyDialer extends Activity implements View.OnClickListener,
 
         mEmergencyActionGroup = (EmergencyActionGroup) findViewById(R.id.emergency_action_group);
 
-        if (mAreEmergencyDialerShortcutsEnabled) {
-            mEccInfoHelper = new EccInfoHelper(new IsoToEccProtobufRepository());
-            setupEmergencyShortcutsView();
-        }
+        mEccInfoHelper = new EccInfoHelper(new IsoToEccProtobufRepository());
+        setupEmergencyShortcutsView();
     }
 
     @Override
@@ -498,8 +478,7 @@ public class EmergencyDialer extends Activity implements View.OnClickListener,
         // If emergency dialer shortcut is enabled and Dialpad view is visible, pressing the
         // back key will back to display EmergencyShortcutView view.
         // Otherwise, it would finish the activity.
-        if (mAreEmergencyDialerShortcutsEnabled && mDialpadView != null
-                && mDialpadView.getVisibility() == View.VISIBLE) {
+        if (mDialpadView != null && mDialpadView.getVisibility() == View.VISIBLE) {
             switchView(mEmergencyShortcutView, mDialpadView, true);
             return;
         }
@@ -716,21 +695,10 @@ public class EmergencyDialer extends Activity implements View.OnClickListener,
         mUserActions = MetricsWriter.USER_ACTION_NONE;
         mMetricsWriter.writeMetricsForEnter();
 
-        // It does not support dark text theme, when emergency dialer shortcuts are enabled.
-        // And set background color to black.
-        if (mAreEmergencyDialerShortcutsEnabled) {
-            mBackgroundGradient.setColors(Color.BLACK, Color.BLACK, false);
-            updateTheme(false);
-        } else {
-            mColorExtractor.addOnColorsChangedListener(this);
-            GradientColors lockScreenColors = mColorExtractor.getColors(WallpaperManager.FLAG_LOCK,
-                    ColorExtractor.TYPE_EXTRA_DARK);
-            // Do not animate when view isn't visible yet, just set an initial state.
-            mBackgroundGradient.setColors(lockScreenColors, false);
-            updateTheme(lockScreenColors.supportsDarkText());
-        }
+        mBackgroundGradient.setColors(Color.BLACK, Color.BLACK, false);
+        updateTheme(false);
 
-        if (mAreEmergencyDialerShortcutsEnabled && mEccInfoHelper != null) {
+        if (mEccInfoHelper != null) {
             final Context context = this;
             mEccInfoHelper.getCountryEccInfoAsync(context,
                     new EccInfoHelper.CountryEccInfoResultCallback() {
-- 
2.17.1

